src/v/security/gssapi_authenticator.cc

```diff
@@ -0,0 +1,514 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#include "security/gssapi_authenticator.h"
+
+#include "bytes/bytes.h"
+#include "config/configuration.h"
+#include "kafka/protocol/request_reader.h"
+#include "security/acl.h"
+#include "security/errc.h"
+#include "security/gssapi.h"
+#include "security/logger.h"
+#include "vlog.h"
+
+#include <boost/outcome/basic_outcome.hpp>
+#include <boost/outcome/success_failure.hpp>
+#include <fmt/ranges.h>
+#include <gssapi/gssapi.h>
+#include <gssapi/gssapi_ext.h>
+
+#include <array>
+#include <sstream>
+#include <string_view>
+
+namespace security {
+
+std::ostream&
+operator<<(std::ostream& os, gssapi_authenticator::state const s) {
+    using state = gssapi_authenticator::state;
+    switch (s) {
+    case state::init:
+        return os << "init";
+    case state::more:
+        return os << "more";
+    case state::ssfcap:
+        return os << "ssfcap";
+    case state::ssfreq:
+        return os << "ssfreq";
+    case state::complete:
+        return os << "complete";
+    case state::failed:
+        return os << "failed";
+    }
+}
+
+static void display_status_1(std::string_view m, OM_uint32 code, int type) {
+    while (true) {
+        OM_uint32 msg_ctx{0};
+        OM_uint32 min_stat{};
+        gss::buffer msg;
+        auto maj_stat = gss_display_status(
+          &min_stat, code, type, GSS_C_NO_OID, &msg_ctx, &msg);
```

Comment:
Do we care of msg_ctx != 0?  I believe, if I'm reading the docs correctly (https://docs.oracle.com/cd/E19455-01/816-3322/6m9k32d10/index.html) a non-0 return here indicates there may be more info to get back?

Reply:
Valid question. Much of this is copy-pasta from examples online. It's my understanding that it's not relevant for the use case of the plugins we're using, but it's a touch under-documented.

Reply:
Yeah I can see the challenge you went through for this

Reply:
I'd prefer to have an actually nice API and a `fmt::formatter` specialisation for a return type/exception/result that contained the major,minor pair.

Reply:
i think there is a decent example of this API being used in Ceph, fwiw.

Reply:
Per the man page:
> gss_display_status() will return a non-zero value in this parameter if there are further messages to extract

So yes, if `gss_display_status` sets `msg_ctx` to a non-0 number, there is more status information to extract.